qiskit_ibm_runtime/utils/json.py

```diff
@@ -243,12 +244,21 @@ def default(self, obj: Any) -> Any:  # pylint: disable=arguments-differ
             )
             return {"__type__": "Parameter", "__value__": value}
         if isinstance(obj, ParameterExpression):
-            value = _serialize_and_encode(
-                data=obj,
-                serializer=_write_parameter_expression,
-                compress=False,
-                use_symengine=bool(optionals.HAS_SYMENGINE),
-            )
+            if QPY_VERSION >= 12:
+                value = _serialize_and_encode(
+                    data=obj,
+                    serializer=_write_parameter_expression,
+                    compress=False,
+                    use_symengine=bool(optionals.HAS_SYMENGINE),
+                    version=QPY_VERSION,
```

Comment:
This version wants to be compatible (less than) whatever version of QPY you'll be using on the backend to deserialise - `QPY_VERSION` will always be the max the local version of Qiskit can handle.

Reply:
Is the use of `QPY_VERSION` incorrect here then? I don't believe the server side has updated to qiskit 1.1 yet

Reply:
You should repeat the logic in other places around the version kwarg and keep the version there the same as in other places. It might makes sense to extract that to a constant like `RUNTIME_QPY_VERSION = 11` or something.

```
if (_TERRA_VERSION[0] == 1 and _TERRA_VERSION[1] >= 1) or _TERRA_VERSION[0] > 1:
    kwargs["version"] = 11
```